libtsuba/src/RDGCore.cpp

```diff
@@ -98,7 +100,11 @@ AddProperties(
           tsuba::ErrorCode::Exists, "column names are not distinct");
     }
   }
-  return UpsertProperties(props, to_update, prop_state);
+  auto written_prop_names =
+      KATANA_CHECKED(UpsertProperties(props, to_update, prop_state));
+
+  // TODO (yige): return write set to TxnContext
```

Comment:
Why leave this as a `TODO`?

Reply:
I'll add it later in this PR if people think the current way of doing it looks good. Or it can be pushed in a follow-up PR.

Reply:
@witchel I'll ei

> Oh another question: Why do only the upserts take a transaction context? Is it because they are non-atomic? However if we want to have multiple inserts or updates we want to have atomic we still need to be able to include them in transactions. Am I missing something?

@arthurp This will go in in the following commits or a followup PR if people think the current way of handling `RDGCore::UpsertProperties` looks fine. It should be very easy since a lot of presumable changes overlap with the current code.